### PR TITLE
Making update mechanism more robust

### PR DESCRIPTION
**Pull request by tsagadar**
There was an edge case when deleting software/rollout before the last device reported back.

Second improvement: use API to get logs in test and don't directly access db.

### CODE REVIEW

File: tests/updater/controller/v1/test_routes.py [hunk 1]
```diff
@@ -283,9 +290,11 @@ async def test_up_to_date(async_client, test_data):
     await _poll(async_client, device.uuid, None, False)
 
 
-async def _assertLogLines(device, expected_line_count):
-    persisted_device = await Device.get(uuid=device.uuid)
-    log = persisted_device.last_log
+async def _assertLogLines(async_client, device, expected_line_count):
```
Comment (b-rowan) [inline tests/updater/controller/v1/test_routes.py:293]: This should be named `_assert_log_lines` to keep with Python naming conventions.

**Review by b-rowan — CHANGES_REQUESTED** [review]

File: tests/updater/controller/v1/test_routes.py (hunk 1 above)
Comment (tsagadar) [reply tests/updater/controller/v1/test_routes.py]: Changed.

**Review by b-rowan — APPROVED** [review]
